src/lib.rs

```diff
@@ -0,0 +1,1012 @@
+			peer_store,
+		}
+	}
+}
+
+/// Wraps all objects that need to be preserved during the run time of [`LdkLite`]. Will be dropped
+/// upon [`LdkLite::stop()`].
+struct Runtime {
+	tokio_runtime: tokio::runtime::Runtime,
+	_background_processor: BackgroundProcessor,
+	stop_networking: Arc<AtomicBool>,
+	stop_wallet_sync: Arc<AtomicBool>,
+}
+
+/// The main interface object of the simplified API, wrapping the necessary LDK and BDK functionalities.
+///
+/// Needs to be initialized and instantiated through [`Builder::build`].
+pub struct LdkLite {
+	running: RwLock<Option<Runtime>>,
+	config: Arc<Config>,
+	chain_access: Arc<ChainAccess<bdk::sled::Tree>>,
+	event_queue: Arc<EventQueue<FilesystemPersister>>,
+	channel_manager: Arc<ChannelManager>,
+	chain_monitor: Arc<ChainMonitor>,
+	peer_manager: Arc<PeerManager>,
+	keys_manager: Arc<KeysManager>,
+	gossip_sync: Arc<GossipSync>,
+	persister: Arc<FilesystemPersister>,
+	logger: Arc<FilesystemLogger>,
+	scorer: Arc<Mutex<Scorer>>,
+	invoice_payer: Arc<InvoicePayer<EventHandler>>,
+	inbound_payments: Arc<PaymentInfoStorage>,
+	outbound_payments: Arc<PaymentInfoStorage>,
+	peer_store: Arc<PeerInfoStorage<FilesystemPersister>>,
+}
+
+impl LdkLite {
+	/// Starts the necessary background tasks, such as handling events coming from user input,
+	/// LDK/BDK, and the peer-to-peer network. After this returns, the [`LdkLite`] instance can be
+	/// controlled via the provided API methods in a thread-safe manner.
+	pub fn start(&mut self) -> Result<(), Error> {
+		// Acquire a run lock and hold it until we're setup.
+		let mut run_lock = self.running.write().unwrap();
+		if run_lock.is_some() {
+			// We're already running.
+			return Err(Error::AlreadyRunning);
+		}
+
+		let runtime = self.setup_runtime()?;
+		*run_lock = Some(runtime);
+		Ok(())
+	}
+
+	/// Disconnects all peers, stops all running background tasks, and shuts down [`LdkLite`].
+	pub fn stop(&mut self) -> Result<(), Error> {
+		let mut run_lock = self.running.write().unwrap();
+		if run_lock.is_none() {
+			return Err(Error::NotRunning);
+		}
+
+		let runtime = run_lock.as_ref().unwrap();
+
+		// Stop wallet sync
+		runtime.stop_wallet_sync.store(true, Ordering::Release);
+
+		// Stop networking
+		runtime.stop_networking.store(true, Ordering::Release);
+		self.peer_manager.disconnect_all_peers();
+
+		// Drop the runtime, which stops the background processor and any possibly remaining tokio threads.
+		*run_lock = None;
+		Ok(())
+	}
+
+	fn setup_runtime(&self) -> Result<Runtime, Error> {
+		let tokio_runtime =
+			tokio::runtime::Builder::new_multi_thread().enable_all().build().unwrap();
+
+		// Setup wallet sync
+		let chain_access = Arc::clone(&self.chain_access);
+		let sync_cman = Arc::clone(&self.channel_manager);
+		let sync_cmon = Arc::clone(&self.chain_monitor);
+		let sync_logger = Arc::clone(&self.logger);
+		let stop_wallet_sync = Arc::new(AtomicBool::new(false));
+		let stop_sync = Arc::clone(&stop_wallet_sync);
+
```

Comment:
Noted, opened https://github.com/lightningdevkit/ldk-lite/issues/29 for this. My approach will likely be to check that all confirmables are on the same page in `chain_access.sync` and otherwise enter a recovery process there.